source/common/filesystem/inotify/watcher_impl.cc

```diff
@@ -34,9 +34,9 @@ WatcherImpl::~WatcherImpl() { close(inotify_fd_); }
 void WatcherImpl::addWatch(absl::string_view path, uint32_t events, OnChangedCb callback) {
   // Because of general inotify pain, we always watch the directory that the file lives in,
   // and then synthetically raise per file events.
-  auto result = api_.fileSystem().splitPathFromFilename(path);
-  auto directory = result.first;
-  auto file = std::string(result.second);
+  PathSplitResult result = api_.fileSystem().splitPathFromFilename(path);
+  auto directory = result.directory_;
+  std::string file = std::string(result.file_);
```

Comment:
const, or just do inline if this is only referenced in one place. Same comment elsewhere in this diff.